src/base/memory_cwrapper.h

```diff
@@ -135,6 +135,13 @@ cub_realloc (void *ptr, size_t size, const char *file, const int line)
 
   if (mmon_is_memory_monitor_enabled ())
     {
+      /* Realloc spec.: Realloc should work same as malloc if ptr == NULL */
+      if (ptr == NULL)
+	{
+	  return cub_alloc (size, file, line);
+	}
+
+      /* Realloc spec.: If input size is zero, just free ptr */
       if (size == 0)
```

Comment:
About size == 0 case, the [manual](https://man7.org/linux/man-pages/man3/realloc.3p.html) define as the following:

> If size is 0, either:
> *  A null pointer shall be returned and, if ptr is not a null
>    pointer, [errno](https://man7.org/linux/man-pages/man3/errno.3.html) shall be set to an implementation-defined
>    value.
> *  A pointer to the allocated space shall be returned, and the
>    memory object pointed to by ptr shall be freed. The
>    application shall ensure that the pointer is not used to
>    access an object.

We must follow one or the other behavior. What specification did you refer to for size==0 case?

Reply:
I referred to MSDN spec. I'll change it.